Constantly getting "Newpipe isn't responding" popups randomly while watching Youtube videos

### Checklist

- [X] I am able to reproduce the bug with the [latest version](https://github.com/TeamNewPipe/NewPipe/releases/latest).
- [X] I made sure that there are *no existing issues* - [open](https://github.com/TeamNewPipe/NewPipe/issues) or [closed](https://github.com/TeamNewPipe/NewPipe/issues?q=is%3Aissue+is%3Aclosed) - which I could contribute my information to.
- [X] I have read the [FAQ](https://newpipe.net/FAQ/) and my problem isn't listed.
- [X] I have taken the time to fill in all the required details. I understand that the bug report will be dismissed otherwise.
- [X] This issue contains only one bug.
- [X] I have read and understood the [contribution guidelines](https://github.com/TeamNewPipe/NewPipe/blob/dev/.github/CONTRIBUTING.md).

### Affected version

v0.24.1

### Steps to reproduce the bug

This issue randomly occurs for me.
Steps to reproduce:
1. Open and play a Youtube video or a few Youtube videos
2. Watch videos until the popup occurs
3. Popup occurs again a few seconds after closing it with the wait button and keeps occurring every time it is closed by the wait button

### Expected behavior

The expected behavior is for the popup to show up when the app is actually unresponsive. The app is fully responsive right before the popup shows up and whenever I close the popup the app is still fully responsive after. 

### Actual behavior

The "Newpipe isn't responding" keeps appearing randomly whenever I am watching Youtube videos with Newpipe. I haven't found anyway to reliable reproduce the issue but it occurs randomly for me whenever I am just watching Youtube videos. 

### Screenshots/Screen recordings

![image](https://user-images.githubusercontent.com/95009201/201498405-eb6559ee-23b6-47dd-9b6b-ede5bbbb8943.png)


### Logs

_No response_

### Affected Android/Custom ROM version

Android 13 stable

### Affected device model

Google Pixel 5a

### Additional information
I found this bug occurring in the official Newpipe app that is run by the Newpipe team.

Okay managed to isolate the relevant lines.

PID: 7460 is NewPipe heer :)

### The Output always generated hen "Not Responding" pops up:
```
06-26 14:03:21.945  1793  2172 D CoreBackPreview: Window{744d57b u0 Application Not Responding: org.schabi.newpipe}: Setting back callback OnBackInvokedCallbackInfo{mCallback=android.window.WindowOnBackInvokedDispatcher$OnBackInvokedCallbackWrapper@fb36f62, mPriority=0}
06-26 14:03:21.973  1793  7460 E OpenGLRenderer: Unable to match the desired swap behavior.
06-26 14:03:22.019  1793  7460 W Parcel  : Expecting binder but got null!
```


### When I press Close on the popup:
```
06-26 14:03:26.338  1058  6951 D C2BqBuffer: cannot dequeue buffer 0
06-26 14:03:26.338  1058  6951 E QC2Buf  : allocate: failed to alloc graphic block w(1280)/h(720)/f(7fa30c06)/usage(30010000)
06-26 14:03:26.339  1402 27949 D BufferPoolAccessor2.0: bufferpool2 0xb400007605c41ff8 : 4(16384 size) total buffers - 0(0 size) used buffers - 3023/3030 (recycle/alloc) - 17/3030 (fetch/transfer)
06-26 14:03:26.339  1793  2907 D MediaSessionService: Media button session is changed to null
06-26 14:03:26.340   998 11586 D msm8974_platform: platform_set_channel_map mixer_ctl_name:Playback Channel Map9
06-26 14:03:26.340   998 11586 D msm8974_platform: platform_set_channel_map: set mapping(1 2 0 0 0 0 0 0) for channel:2
06-26 14:03:26.340  1793  5028 I WindowManager: WIN DEATH: Window{868fc16 u0 org.schabi.newpipe/org.schabi.newpipe.MainActivity}
06-26 14:03:26.341   991   991 I Zygote  : Process 4580 exited due to signal 9 (Killed)
06-26 14:03:26.341  1793  5028 W InputManager-JNI: Input channel object '868fc16 org.schabi.newpipe/org.schabi.newpipe.MainActivity (client)' was disposed without first being removed with the input manager!
06-26 14:03:26.345 27793 27793 I MediaPlayerList: onMediaKeyEventSessionChanged: packageName=org.schabi.newpipe
06-26 14:03:26.348 27793 27793 W MediaPlayerList: org.schabi.newpipe is already the active player
06-26 14:03:26.348 27793 27793 V MediaPlayerList: onActiveSessionsChanged: number of controllers: 0
06-26 14:03:26.348 27793 27793 W AudioMediaPlayerWrapper: The session was destroyed org.schabi.newpipe
06-26 14:03:26.349  1058  6951 W QC2V4l2Codec: [avcD_227] Graphic output-buf allocation failed!
06-26 14:03:26.349  1058  6951 E QC2V4l2Codec: [avcD_227] Error during buffer allocation
06-26 14:03:26.350  1058  6951 D C2BqBuffer: cannot dequeue buffer 0
06-26 14:03:26.350  1058  6947 E QC2Comp : [avcD_227][ERROR] entered
06-26 14:03:26.350  1058  6951 E QC2Buf  : allocate: failed to alloc graphic block w(1280)/h(720)/f(7fa30c06)/usage(30010000)
06-26 14:03:26.350  1058  6951 W QC2V4l2Codec: [avcD_227] Graphic output-buf allocation failed!
06-26 14:03:26.350  1058  6951 E QC2V4l2Codec: [avcD_227] Error during buffer allocation
06-26 14:03:26.350  1058  6951 D C2BqBuffer: cannot dequeue buffer 0
06-26 14:03:26.350  1058  6951 E QC2Buf  : allocate: failed to alloc graphic block w(1280)/h(720)/f(7fa30c06)/usage(30010000)
06-26 14:03:26.350  1058  6951 W QC2V4l2Codec: [avcD_227] Graphic output-buf allocation failed!
06-26 14:03:26.350  1058  6951 E QC2V4l2Codec: [avcD_227] Error during buffer allocation
06-26 14:03:26.350  1058  6951 D C2BqBuffer: cannot dequeue buffer 0
```
The last three lines repeat for some more time.

--- a/app/src/main/java/org/schabi/newpipe/player/PlayerService.java
+++ b/app/src/main/java/org/schabi/newpipe/player/PlayerService.java
@@ -29,6 +29,7 @@ import android.os.IBinder;
 import android.util.Log;
 
 import org.schabi.newpipe.player.mediasession.MediaSessionPlayerUi;
+import org.schabi.newpipe.player.notification.NotificationPlayerUi;
 import org.schabi.newpipe.util.ThemeHelper;
 
 import java.lang.ref.WeakReference;
@@ -59,6 +60,14 @@ public final class PlayerService extends Service {
         ThemeHelper.setTheme(this);
 
         player = new Player(this);
+        /*
+        Create the player notification and start immediately the service in foreground,
+        otherwise if nothing is played or initializing the player and its components (especially
+        loading stream metadata) takes a lot of time, the app would crash on Android 8+ as the
+        service would never be put in the foreground while we said to the system we would do so
+         */
+        player.UIs().get(NotificationPlayerUi.class)
+                .ifPresent(NotificationPlayerUi::createNotificationAndStartForeground);
     }
 
     @Override
@@ -68,16 +77,38 @@ public final class PlayerService extends Service {
                     + "], flags = [" + flags + "], startId = [" + startId + "]");
         }
 
+        /*
+        Be sure that the player notification is set and the service is started in foreground,
+        otherwise, the app may crash on Android 8+ as the service would never be put in the
+        foreground while we said to the system we would do so
+        The service is always requested to be started in foreground, so always creating a
+        notification if there is no one already and starting the service in foreground should
+        not create any issues
+        If the service is already started in foreground, requesting it to be started shouldn't
+        do anything
+         */
+        if (player != null) {
+            player.UIs().get(NotificationPlayerUi.class)
+                    .ifPresent(NotificationPlayerUi::createNotificationAndStartForeground);
+        }
+
         if (Intent.ACTION_MEDIA_BUTTON.equals(intent.getAction())
-                && player.getPlayQueue() == null) {
-            // No need to process media button's actions if the player is not working, otherwise the
-            // player service would strangely start with nothing to play
+                && (player == null || player.getPlayQueue() == null)) {
+            /*
+            No need to process media button's actions if the player is not working, otherwise
+            the player service would strangely start with nothing to play
+            Stop the service in this case, which will be removed from the foreground and its
+            notification cancelled in its destruction
+             */
+            stopSelf();
             return START_NOT_STICKY;
         }
 
-        player.handleIntent(intent);
-        player.UIs().get(MediaSessionPlayerUi.class)
-                .ifPresent(ui -> ui.handleMediaButtonIntent(intent));
+        if (player != null) {
+            player.handleIntent(intent);
+            player.UIs().get(MediaSessionPlayerUi.class)
+                    .ifPresent(ui -> ui.handleMediaButtonIntent(intent));
+        }
 
         return START_NOT_STICKY;
     }
@@ -87,7 +118,7 @@ public final class PlayerService extends Service {
             Log.d(TAG, "stopForImmediateReusing() called");
         }
 
-        if (!player.exoPlayerIsNull()) {
+        if (player != null && !player.exoPlayerIsNull()) {
             // Releases wifi & cpu, disables keepScreenOn, etc.
             // We can't just pause the player here because it will make transition
             // from one stream to a new stream not smooth
@@ -98,7 +129,7 @@ public final class PlayerService extends Service {
     @Override
     public void onTaskRemoved(final Intent rootIntent) {
         super.onTaskRemoved(rootIntent);
-        if (!player.videoPlayerSelected()) {
+        if (player != null && !player.videoPlayerSelected()) {
             return;
         }
         onDestroy();

--- a/app/src/main/java/org/schabi/newpipe/player/notification/NotificationPlayerUi.java
+++ b/app/src/main/java/org/schabi/newpipe/player/notification/NotificationPlayerUi.java
@@ -17,21 +17,11 @@ import org.schabi.newpipe.player.helper.PlayerHelper;
 import org.schabi.newpipe.player.ui.PlayerUi;
 
 public final class NotificationPlayerUi extends PlayerUi {
-    private boolean foregroundNotificationAlreadyCreated = false;
     private final NotificationUtil notificationUtil;
 
     public NotificationPlayerUi(@NonNull final Player player) {
         super(player);
         notificationUtil = new NotificationUtil(player);
-    }
-
-    @Override
-    public void initPlayer() {
-        super.initPlayer();
-        if (!foregroundNotificationAlreadyCreated) {
-            notificationUtil.createNotificationAndStartForeground();
-            foregroundNotificationAlreadyCreated = true;
-        }
     }
 
     @Override
@@ -121,5 +111,9 @@ public final class NotificationPlayerUi extends PlayerUi {
     public void onPlayQueueEdited() {
         super.onPlayQueueEdited();
         notificationUtil.createNotificationIfNeededAndUpdate(false);
+    }
+
+    public void createNotificationAndStartForeground() {
+        notificationUtil.createNotificationAndStartForeground();
     }
 }

